superset-frontend/src/dashboard/components/nativeFilters/FilterBar/index.tsx

```diff
@@ -200,7 +205,9 @@ const publishDataMask = debounce(
     } else {
       dataMaskKey = await createFilterKey(dashboardId, dataMask, tabId);
     }
-    newParams.set(URL_PARAMS.nativeFiltersKey.name, dataMaskKey);
+    if (dataMaskKey) {
```

Comment:
When this is not defined?

Reply:
If I understand this correctly, when navigating to the dashboard without a native filter key, this will be undefined (this is later populated when the current state is cached)